sanic/app.py

```diff
@@ -104,31 +144,33 @@ def __init__(
         if configure_logging:
             logging.config.dictConfig(log_config or LOGGING_CONFIG_DEFAULTS)
 
-        self.name = name
+        self._asgi_client = None
+        self._blueprint_order: List[Blueprint] = []
+        self._test_client = None
+        self._test_manager = None
         self.asgi = False
-        self.router = router or Router()
-        self.signal_router = signal_router or SignalRouter()
-        self.request_class = request_class
-        self.error_handler = error_handler or ErrorHandler()
-        self.config = Config(load_env=load_env)
-        self.request_middleware: Deque[MiddlewareType] = deque()
-        self.response_middleware: Deque[MiddlewareType] = deque()
         self.blueprints: Dict[str, Blueprint] = {}
-        self._blueprint_order: List[Blueprint] = []
+        self.config = Config(load_env=load_env)
         self.configure_logging = configure_logging
+        self.ctx = SimpleNamespace()
```

Comment:
This could later be an option, like the `router`, such as:

```python
self.ctx = ctx or SimpleNamespace()
```

No need to do that now, but I would like to open that discussion for later :wink:

Reply:
I think this is actually sort of unnecessary. The whole purpose of the context object, and using `SimpleNamespace` in particular is that you can set anything you want on it.